src/Controls/src/Core/Shapes/Geometry.cs

```diff
@@ -1,7 +1,9 @@
+using Microsoft.Maui.Graphics;
+
 namespace Microsoft.Maui.Controls.Shapes
 {
-	public abstract class Geometry : BindableObject
+	public abstract class Geometry : BindableObject, IShape
 	{
-
+		public abstract PathF PathForBounds(Graphics.Rectangle rect);
```

Comment:
I'm not convinced this is the right approach.  Consider a Line with absolute Start/End Points.  Since the Geometry itself has no concept of a width/height, there's no way to know how to scale the start/end points and the line itself to the requested rectangle to return the path for.

In practice, on Geometry, PathForBounds is always passed the View's bounds that will display the Geometry so this should _work_ but it's rather confusing.

Maybe a `public abstract void AppendPath(PathF path)` would be more descriptive.

Reply:
Switched to this.